.github/workflows/_sitrep_mgmn.yaml

```diff
@@ -0,0 +1,116 @@
+
+jobs:
+  sitrep:
+    runs-on: ubuntu-22.04
+    env:
+      BADGE_FILENAME_FULL: ${{ inputs.BADGE_FILENAME }}.json
+    outputs:
+      STATUS: ${{ steps.gen-sitrep.outputs.STATUS }}
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Download all artifacts from the previous jobs
+        uses: actions/download-artifact@v3
+
+      - name: Write Test Status
+        shell: bash -x -e {0}
+        run: |
+          EXIT_STATUSES="${GITHUB_RUN_ID}-*/*-status.json"
+
+          cat <<EOF >>$GITHUB_STEP_SUMMARY
+          ## T5x MGMN+SPMD Test Status
+          | Test Case | State | Exit Code |
+          | --- | --- | --- |
+          EOF
+          for i in $EXIT_STATUSES; do
+            # Files are named <GHID>-<NAME>/<NAME>-status.json
+            echo "| $(echo $i | cut -d/ -f1 | cut -d- -f2) | $(jq -r .state $i) | $(jq -r .exitcode $i)"
+          done | tee -a $GITHUB_STEP_SUMMARY
+
+          echo "Test statuses:"
+          jq -rc 'input_filename,.' $EXIT_STATUSES
+
+      - name: Generate sitrep
+        id: gen-sitrep
+        shell: bash -x -e {0}
+        run: |
+          source .github/workflows/scripts/to_json.sh
+
+          EXIT_STATUSES="${GITHUB_RUN_ID}-*/*-status.json"
+          passed_tests=$(jq -r '. | select ((.state == "COMPLETED") and (.exitcode == "0")) | .state' $EXIT_STATUSES | wc -l)
+          failed_tests=$(jq -r '. | select ((.state != "COMPLETED") or (.exitcode != "0")) | .state' $EXIT_STATUSES | wc -l)
+          total_tests=$(ls $EXIT_STATUSES | wc -l)
+
+          METRICS_LOG=metrics-test-log/report.jsonl
+          all_outcomes() {
+            cat $METRICS_LOG | jq -r '. | select((.["$report_type"] == "TestReport") and (.when == "call")) | .outcome'
+          }
+          cnt_type() {
+            cat $METRICS_LOG | jq '. | select((.["$report_type"] == "TestReport") and (.when == "call") and (.outcome | contains("'${1}'"))) | .outcome' | wc -l
+          }
+          pytest_failed_tests=$(cnt_type failed)
+          pytest_passed_tests=$(cnt_type passed)
+          pytest_total_tests=$(all_outcomes | wc -l)
+
+          if ([[ $failed_tests -eq 0 ]] && [[ $total_tests -gt 0 ]] && \
+              [[ $pytest_failed_tests -eq 0 ]] && [[ $pytest_total_tests -gt 0 ]]); then
+            status=success
+            badge_color=brightgreen
+            badge_message="${passed_tests}/${total_tests} ran ${pytest_passed_tests}/${pytest_total_tests} pass loss+perf"
+          elif [[ $passed_tests -eq 0 ]] || [[ $pytest_passed_tests -eq 0 ]]; then
+            status=failure
+            badge_color=red
+            badge_message="${passed_tests}/${total_tests} ran ${pytest_passed_tests}/${pytest_total_tests} pass loss+perf"
+          else
+            status=failure
+            badge_color=yellow
+            badge_message='error'
+          fi
+
+          badge_label='Upstream Tests'
+          summary="T5X MGMN Test: $badge_message"
```

Comment:
Should 'T5X' be parameterized by some input variable?

Reply:
Done.